vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -272,25 +272,34 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 	filters.Add("status", "running")
 	filters.Add("status", "paused")
 	filters.Add("status", "restarting")
-	containers, err := c.ContainerList(context.Background(), types.ContainerListOptions{
+
+	ctx, cancel := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
```

Comment:
The cancel function releases the context resources. [from here](https://golang.org/pkg/context/#WithTimeout). So if the docker api (in which we use the context) returns before timeout, the context is cancelled. This context if used again won't timeout again after the specified timeout, because it was cancelled once. Tried this with a small go code to reuse the context created once and used again and again in a loop. So I think we can't reuse the context.